openedx/core/djangoapps/timed_notification/core.py

```diff
@@ -0,0 +1,153 @@
+import logging
+
+from student.models import CourseEnrollment
+from edxmako.shortcuts import render_to_string
+
+from django.core.urlresolvers import reverse
+from django.core.mail import EmailMultiAlternatives, get_connection
+from django.conf import settings
+
+from collections import Counter
+from smtplib import SMTPDataError
+from boto.ses.exceptions import (
+    SESAddressBlacklistedError,
+    SESDomainEndsWithDotError,
+    SESLocalAddressCharacterError,
+    SESIllegalAddressError,
+)
+
+log = logging.getLogger('edx.celery.task')
+
+# Errors that an individual email is failing to be sent, and should just
+# be treated as a fail.
+SINGLE_EMAIL_FAILURE_ERRORS = (
+    SESAddressBlacklistedError,  # Recipient's email address has been temporarily blacklisted.
+    SESDomainEndsWithDotError,  # Recipient's email address' domain ends with a period/dot.
+    SESIllegalAddressError,  # Raised when an illegal address is encountered.
+    SESLocalAddressCharacterError,  # An address contained a control or whitespace character.
+)
+
+
+def send_course_notification_email(course, mako_template_path, context, to_list=None):
+
+    """
+    Sends an email to a list of recipients.
+
+    Inputs are:
+      * `course`: notification about this course.
+      * `mako_template_path`: Mako template path.
+      * `context`: context for template
+      * `to_list`: list of recipients, if list is not provided then the email will be send to all enrolled students.
+
+    """
+
+    log.info("Sending email for course %s", course)
+    if to_list is None:
+        to_list = CourseEnrollment.objects.users_enrolled_in(course.id)
+
+    total_recipients = len(to_list)
+    recipient_num = 0
+    total_recipients_successful = 0
+    total_recipients_failed = 0
+    recipients_info = Counter()
+
+    log.info(
+        "TimedNotification ==> TotalRecipients: %s",
+        total_recipients
+    )
+
+    try:
+        connection = get_connection()
+        connection.open()
+
+        for current_recipient in to_list:
+            recipient_num += 1
+            email = current_recipient.email
+            subject = "Philanthropy U Team Course Notification"
+            context['full_name'] = current_recipient.username
```

Comment:
should not use `username` here. instead `current_recipient.first_name` and `current_recipient.last_name`